addon/services/intl.js

```diff
@@ -52,6 +52,10 @@ const IntlService = Service.extend(Evented, {
   formatTime: formatterProxy('time'),
   formatDate: formatterProxy('date'),
 
+  t(...args) {
+    return this.formatMessage(this.findTranslationByKey(...args));
```

Comment:
`{{t 'foo' bar='bar' locale='fr'}}`

The formats gets threaded through automatically.  But when invoking it programmatically, you have the ability to thread a format POJO.  Ultimately, this gets invoked: https://github.com/yahoo/ember-intl/blob/master/addon/formatters/format-message.js#L21 and that all should be satisfied